tests/MassTransit.Containers.Tests/Stop_Specs.cs

```diff
@@ -0,0 +1,61 @@
+namespace MassTransit.Containers.Tests
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using NUnit.Framework;
+    using TestFramework;
+    using TestFramework.Messages;
+    using Testing;
+
+
+    [TestFixture]
+    public class Stop_Specs :
+        InMemoryTestFixture
+    {
+        [Test]
+        public async Task Should_respect_ConsumerStopTimeout()
+        {
+            var collection = new ServiceCollection();
+            collection.AddOptions<MassTransitHostOptions>().Configure(options => options.ConsumerStopTimeout = TimeSpan.FromSeconds(1));
+            collection.AddMassTransitTestHarness(configurator =>
+            {
+                configurator.AddTaskCompletionSource<ConsumeContext<PingMessage>>();
+                configurator.AddConsumer<StuckConsumer>();
+            });
+
+            var provider = collection.BuildServiceProvider(true);
+
+            var harness = provider.GetTestHarness();
+            await harness.Start();
+
+            await harness.Bus.Publish(new PingMessage(), harness.CancellationToken);
+
+            await provider.GetTask<ConsumeContext<PingMessage>>();
+
+            await harness.Stop(harness.CancellationToken);
+
+            using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(3));
+            Assert.That(await harness.Consumed.Any<PingMessage>(cts.Token), Is.True);
```

Comment:
This should check that exception is not null, since it shoudl have thrown a ConsumerCanceledException

Reply:
`await harness.Consumed.Any<PingMessage>(x => x.Exception != null, cts.Token)`